Lib/threading.py

```diff
@@ -1510,17 +1397,15 @@ def __init__(self):
                         daemon=_daemon_threads_allowed())
         self._started.set()
         self._set_ident()
+        self._handle = _make_thread_handle(self._ident)
         if _HAVE_THREAD_NATIVE_ID:
             self._set_native_id()
         with _active_limbo_lock:
             _active[self._ident] = self
         _DeleteDummyThreadOnDel(self)
 
-    def _stop(self):
-        pass
-
     def is_alive(self):
-        if not self._is_stopped and self._started.is_set():
+        if self._started.is_set() and not self._handle.is_done():
             return True
         raise RuntimeError("thread is not alive")
```

Comment:
I'm assuming `_DummyThread` is sufficiently tested that we aren't risking breaking them somehow with this change.  The note above seems somewhat particular about their behavior, including the part about how we can' t wait for them.

Reply:
I'll flip this back, it just felt more natural to me.